### PR TITLE
Bug-2118970: Create an utility GetIgnitionEndpoint function

### PR DESCRIPTION
**Pull request by mkowalski**
Contributes-to: [MGMTBUGSM-525](https://issues.redhat.com//browse/MGMTBUGSM-525)

### CODE REVIEW

File: internal/host/hostutil/host_utils.go [hunk 1]
```diff
@@ -298,3 +300,20 @@ func IsDiskEncryptionEnabledForRole(encryption models.DiskEncryption, role model
 		return false
 	}
 }
+
+func GetIgnitionEndpoint(cluster *common.Cluster, host *models.Host) (string, error) {
+	poolName := string(common.GetEffectiveRole(host))
+	if host.MachineConfigPoolName != "" {
+		poolName = host.MachineConfigPoolName
+	}
```
Comment (eranco74) [inline internal/host/hostutil/host_utils.go:308]: Can we have a default "worker" here or in GetEffectiveRole?

Comment (slaviered) [reply internal/host/hostutil/host_utils.go:308]: Not in GetEffectiveRole since the default of role is auto-assign

Comment (mkowalski) [reply internal/host/hostutil/host_utils.go:316]: Sure, just added it here. I hope it's a dead code, but we never know

**Pull request merged**